src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -417,13 +417,39 @@ private static object CreateInstance([DynamicallyAccessedMembers(DynamicallyAcce
                 throw new InvalidOperationException(SR.Format(SR.Error_CannotActivateAbstractOrInterface, type));
             }
 
-            if (!type.IsValueType)
+            ConstructorInfo[] constructors = type.GetConstructors(BindingFlags.Public | BindingFlags.Instance);
+
+            bool hasParameterlessConstructor =
+                constructors.Any(ctor => ctor.GetParameters().Length == 0);
```

Comment:
```suggestion
                constructors.Any(ctor => ctor.GetParameters().Length == 0) || type.IsValueType;
```

This fixes the test in https://github.com/dotnet/runtime/pull/67258#issuecomment-1117617842 which we should add but with a more descriptive name. This does cause the new `StructTypeOptions` tests to fail, but I think that's okay because struct options are niche anyway, so we should just endeavor not to break anyone in this case.

The `RecordStructTypeOptions` won't fail though for the same reason these tests currently pass in `main` since `PropertyInfo.SetValue()` works just fine on `init-only` properties. While I doubt this was intentional, at least we don't regress this. We'd still use the constructor for class structs which is preferable.

Reply:
Sorry, I missed the previous comment. I think even though I gave it a thumbs up, I intended to resolve it when time permitted, but then it just popped off the stack!

Have added the test, changed the logic, and removed the failing struct options tests.